lessons/advanced/metaprogramming.md

```diff
@@ -131,6 +131,53 @@ def test do
 end
 ```
 
+## Debugging
+
+Okay, right now we know how to `quote/2`, `unquote/1` and write macroses. But what if you have a huge chunk of quoted code and want to understand it? In this case, you can use `Macro.to_string/2`. Take a look at this example:
+
+```elixir
+iex> Macro.to_string(quote(do: foo.bar(1, 2, 3)))
+"foo.bar(1, 2, 3)"
+```
+
+And when you want to look at the code generated by macros you can also combine this with `Macro.expand/2` and `Macro.expand_once/2`. These functions expand macros in the given quoted code. The first may expand it several times, while the former - only once. For example, if we take slightly modified 'unless' example from the previous section:
```

Comment:
Let's do "you can also combine this with" -> "you can combine them with".

What about: ". These functions" -> ", these functions"